### PR TITLE
[SPARK-42681][SQL] Relax ordering constraint for ALTER TABLE ADD|REPLACE column descriptor

### PR DESCRIPTION
**Pull request by vitaliili-db**
### What changes were proposed in this pull request?

Currently the grammar for ALTER TABLE ADD|REPLACE column is:

```
qualifiedColTypeWithPosition
: name=multipartIdentifier dataType (NOT NULL)? defaultExpression? commentSpec? colPosition?
;
```

This enforces a constraint on the order of: (NOT NULL, DEFAULT value, COMMENT value FIRST|AFTER value). We can update the grammar to allow these options in any order instead, to improve usability.

### Why are the changes needed?

This helps make the SQL syntax more usable.

### Does this PR introduce _any_ user-facing change?

Yes, the SQL syntax updates slightly.

### How was this patch tested?

Existing and new unit tests

### CODE REVIEW

**Comment by vitaliili-db** [comment]
build timed out but succeeded on rerun: https://github.com/vitaliili-db/spark/actions/runs/4346311324/jobs/7598960402

**Comment by vitaliili-db** [comment]
@gengliangwang can you review this please?

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1,4 +1,10 @@
 {
+  "ALTER_TABLE_COLUMN_OPTION_DUPLICATE" : {
```
Comment (gengliangwang) [inline core/src/main/resources/error/error-classes.json:2]: How about 
`ALTER_TABLE_COLUMN_DEFINITION_DUPLICATE`

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
+    "message" : [
+      "ALTER TABLE <type> column <columnName> specifies option \"<optionName>\" more than once, which is invalid."
```
Comment (gengliangwang) [inline core/src/main/resources/error/error-classes.json:4]:
```suggestion
      "ALTER TABLE <type> column <columnName> definition \"<optionName>\" more than once, which is invalid."
```

**Comment by gengliangwang** [comment]
@vitaliili-db Thanks for the work! 
LGTM overall. I got only one comment: the wording "column options" sounds weird. Shall we call it "column descriptor"?
image

**Comment by vitaliili-db** [comment]
@gengliangwang great catch, yes, we should follow standard. Renamed.

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**